File: aws-intro/go/lab-01/code/04-updating-your-infrastructure/step1.go
package main

import (
    "path/filepath"

    "github.com/pulumi/pulumi/sdk/go/pulumi"
    "github.com/pulumi/pulumi-aws/sdk/go/aws/s3"
)

func main() {
    pulumi.Run(func(ctx *pulumi.Context) error {

        bucket, err := s3.NewBucket(ctx, "my-bucket", nil)
        if err != nil {
            return err
        }

        bucketFile := filepath.Join("site", "index.html")
        _, err = s3.NewBucketObject(ctx, "index.html", &s3.BucketObjectArgs{
            Bucket: bucket.Bucket,
            Source: pulumi.NewFileAsset(bucketFile),
        })

        ctx.Export("bucketName", bucket.Bucket)

        return nil
    })
}

File: archive/aws-intro/csharp/lab-01/code/04-updating-your-infrastructure/step1.cs
using System.IO;
using Pulumi;
using Pulumi.Aws.S3;
using Aws = Pulumi.Aws;

class MyStack : Stack
{
    public MyStack()
    {
        var bucket = new Aws.S3.Bucket("my-bucket", new BucketArgs());
        this.BucketName = bucket.BucketName;

        var bucketFile = Path.Combine("site", "index.html");
        var bucketObject = new Aws.S3.BucketObject("index.html", new BucketObjectArgs
        {
            Bucket = bucket.BucketName,
            Source = new FileAsset(bucketFile)
        });
    }

    [Output] public Output<string> BucketName { get; set; }
}
